core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/DbUsernameConverterService.java

```diff
@@ -0,0 +1,47 @@
+package com.sequenceiq.cloudbreak.service.rdsconfig;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DbUsernameConverterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DbUsernameConverterService.class);
+
+    public String toConnectionUsername(String host, String dbUser) {
+        if (DbHostnameMatcher.isAzure(host)) {
+            LOGGER.debug("Detected Azure database server {}, appending short hostname to connection username", host);
+            return dbUser + "@" + StringUtils.substringBefore(host, ".");
+        }
+
+        return dbUser;
+    }
+
+    /**
+     * Converts a connection username to a db username.
+     * On azure, connection username follows the 'username@short-hostname' pattern.
+     * On postgres, however, users have to be in 'username' format, without the '@' sign => code has to take username.
+     *
+     * @param username a connection username
+     * @return returns the username part before the '@' sign
+     */
+    public String toDatabaseUsername(String username) {
+        return username.contains("@") ? StringUtils.substringBefore(username, "@") : username;
```

Comment:
I think substringBefore will return the original string if it does not contain the separator so it's unnecessary to check before.

Reply:
You are right ... Thanks!